spec/examples/schema_adapter_examples.rb

```diff
@@ -120,10 +126,12 @@
       subject.switch(schema1) do
         expect(connection.schema_search_path).to start_with %("#{schema1}")
         expect(User.sequence_name).to eq "#{schema1}.#{User.table_name}_id_seq"
+        expect(Company.sequence_name).to eq "#{public_schema}.#{Company.table_name}_id_seq"
       end
 
       expect(connection.schema_search_path).to start_with %("#{public_schema}")
-      expect(User.sequence_name).to eq "#{public_schema}.#{User.table_name}_id_seq"
+      expect(User.sequence_name).to eq "#{User.table_name}_id_seq"
```

Comment:
And this test currently fails because of my previous note. the sequence name for users is set inside the block and cached, therefore then it's giving the wrong one outside the block

Reply:
Ignore my comment. I found that the issue was that we were initializing an adapter instance by calling `Apartment::Tenant.postgresql_adapter config` but not storing it in the thread current variable. and then when accessing Apartment::Tenant methods, a new adapter was being initialized thus the apartment tenant current was different